cli/src/commit_templater.rs

```diff
@@ -1671,3 +1730,83 @@ fn builtin_tree_diff_methods<'repo>() -> CommitTemplateBuildMethodFnMap<'repo, T
     // TODO: add files() or map() to support custom summary-like formatting?
     map
 }
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct CryptographicSignature {
+    commit: Commit,
+}
+
+impl CryptographicSignature {
+    /// Panics if the commit is not signed.
+    fn new(commit: Commit) -> Self {
+        if !commit.is_signed() {
+            panic!("commit must be signed");
+        }
+        Self { commit }
+    }
+
+    /// Panics if the commit has no signature.
+    fn verify(&self) -> SignResult<Verification> {
+        self.commit
+            .verification()
+            .transpose()
+            .expect("signature must exist")
+    }
+
+    fn key(&self) -> Option<String> {
+        match self.commit.verification() {
+            Ok(Some(verification)) => verification.key,
+            _ => None,
+        }
+    }
+
+    fn display(&self) -> Option<String> {
+        match self.commit.verification() {
+            Ok(Some(verification)) => verification.display,
+            _ => None,
+        }
+    }
+}
+
+pub fn builtin_cryptographic_signature_methods<'repo>(
+) -> CommitTemplateBuildMethodFnMap<'repo, CryptographicSignature> {
+    type L<'repo> = CommitTemplateLanguage<'repo>;
+    // Not using maplit::hashmap!{} or custom declarative macro here because
+    // code completion inside macro is quite restricted.
+    let mut map = CommitTemplateBuildMethodFnMap::<CryptographicSignature>::new();
+    map.insert(
+        "verify",
+        |_language, _diagnostics, _build_ctx, self_property, function| {
+            function.expect_no_arguments()?;
+            let out_property = TemplateFunction::new(self_property, |sig| match sig.verify() {
+                Ok(verification) => match verification.status {
+                    SigStatus::Good => Ok("Good".into()),
+                    SigStatus::Bad => Ok("Bad".into()),
+                    SigStatus::Unknown => Ok("Unknown".into()),
```

Comment:
nit: lowercase "good"/"bad"/"unknown" might be better here. We tend to use kebab-case in config-like context.